csv.h

```diff
@@ -968,8 +968,22 @@ template <class T> void parse_float(const char *col, T &x) {
     ++col;
   } else if (*col == '+')
     ++col;
+  
+  if(   (*col == 'N' && *(col + 1) == 'a' && *(col + 2) == 'N') |
+        (*col == 'i' && *(col + 1) == 'n' && *(col + 2) == 'f')) {
+      
+      if(*col == 'N' && is_neg)
+          throw error::no_digit();
+      x = (*col == 'N') ?
+          std::numeric_limits<T>::quiet_NaN() : std::numeric_limits<T>::infinity();
```

Comment:
The reasons why I never implemented Nan parsing is because there is not a single nan value. Floating point nan values carry an error code called payload that is lost when just writing "nan". 

Apparently, at least GCC `std::to_chars` decided to just drop the nan payload and still claim that `std::to_chars` and `std::from_chars` roundtrip, which they obviously do not. You can verify this by checking the output of `std::nan("1")` and `std::nan("2")`. The resulting bit patterns are not the same.